src/umpire/strategy/AllocationStrategy.hpp

```diff
@@ -51,7 +52,11 @@ class AllocationStrategy {
    *
    * \param ptr Pointer to free.
    */
-  virtual void deallocate(void* ptr) = 0;
+  virtual void deallocate(void* ptr, std::size_t size=0) = 0;
+
+  void* allocate_internal(std::size_t bytes);
```

Comment:
Any thoughts on the naming of these internal/impl methods?

Reply:
Can they be hiddning behind pimpl?